gtk4/src/subclass/entry_buffer.rs

```diff
@@ -0,0 +1,233 @@
+    }
+
+    fn inserted_text(&self, entry_buffer: &Self::Type, position: u32, chars: &str) {
+        self.parent_inserted_text(entry_buffer, position, chars)
+    }
+}
+
+pub trait EntryBufferImplExt: ObjectSubclass {
+    fn parent_delete_text(&self, entry_buffer: &Self::Type, position: u32, n_chars: u32) -> u32;
+    fn parent_deleted_text(&self, entry_buffer: &Self::Type, position: u32, n_chars: u32);
+    fn parent_get_length(&self, entry_buffer: &Self::Type) -> u32;
+    fn parent_get_text(&self, entry_buffer: &Self::Type, n_bytes: &mut usize) -> GString;
+    fn parent_insert_text(&self, entry_buffer: &Self::Type, position: u32, chars: &str) -> u32;
+    fn parent_inserted_text(&self, entry_buffer: &Self::Type, position: u32, chars: &str);
+}
+
+impl<T: EntryBufferImpl> EntryBufferImplExt for T {
+    fn parent_delete_text(&self, entry_buffer: &Self::Type, position: u32, n_chars: u32) -> u32 {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkEntryBufferClass;
+            let f = (*parent_class)
+                .delete_text
+                .expect("No parent class impl for \"delete_text\"");
+            f(
+                entry_buffer
+                    .unsafe_cast_ref::<EntryBuffer>()
+                    .to_glib_none()
+                    .0,
+                position,
+                n_chars,
+            )
+        }
+    }
+
+    fn parent_deleted_text(&self, entry_buffer: &Self::Type, position: u32, n_chars: u32) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkEntryBufferClass;
+            if let Some(f) = (*parent_class).deleted_text {
+                f(
+                    entry_buffer
+                        .unsafe_cast_ref::<EntryBuffer>()
+                        .to_glib_none()
+                        .0,
+                    position,
+                    n_chars,
+                )
+            }
+        }
+    }
+
+    fn parent_get_length(&self, entry_buffer: &Self::Type) -> u32 {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkEntryBufferClass;
+            let f = (*parent_class)
+                .get_length
+                .expect("No parent class impl for \"get_length\"");
+            f(entry_buffer
+                .unsafe_cast_ref::<EntryBuffer>()
+                .to_glib_none()
+                .0)
+        }
+    }
+
+    fn parent_get_text(&self, entry_buffer: &Self::Type, n_bytes: &mut usize) -> GString {
+        unsafe {
+            let data = T::type_data();
+            let parent_class = data.as_ref().get_parent_class() as *mut ffi::GtkEntryBufferClass;
+            let f = (*parent_class)
+                .get_text
+                .expect("No parent class impl for \"get_text\"");
+            from_glib_none(f(
```

Comment:
Use `FromGlibContainer::from_glib_none_num()` here.

```rust
let mut n_bytes = 0;
let res = f(..., &n_bytes);
FromGlibContainer::from_glib_none_num(res, n_bytes as usize)
```

And only return a `GString` and get rid of the `&mut` parameter. It's for the length of the text AFAIU, and that's part of the string anyway...

Reply:
This doesn't seem to work the trait `glib::translate::FromGlibContainer<_, *const i8>` is not implemented for `glib::GString`

Reply:
Then you need to implement it :) The problem is that you must only create a string for given length here, not the whole thing until the NUL terminator